lib/completions/prompt.rb

```diff
@@ -8,13 +8,19 @@ class Prompt
       attr_reader :system_message, :messages
       attr_accessor :tools
 
-      def initialize(system_msg, messages: [], tools: [])
+      def initialize(system_msg = nil, messages: [], tools: [])
         raise ArgumentError, "messages must be an array" if !messages.is_a?(Array)
         raise ArgumentError, "tools must be an array" if !tools.is_a?(Array)
 
-        system_message = { type: :system, content: system_msg }
+        @messages = []
+
+        if system_msg
+          system_message = { type: :system, content: system_msg }
```

Comment:
I think this is setting the `system_message` defined by `attr_reader`; in this case it might be slightly clearer to do `@system_message = `, I was confused otherwise why there was an intermediate var here

Reply:
oh yeah cleaning this up